File: MVVMjava/app/src/main/java/com/example/mvvmjava/model/UserViewModel.java
package com.example.mvvmjava.model;

import androidx.databinding.BaseObservable;
import androidx.databinding.Bindable;
import androidx.databinding.ObservableField;

import com.example.mvvmjava.BR;

public class UserViewModel extends BaseObservable {

    private String name;
    private int num = 0;

    public ObservableField<String> observableField = new ObservableField<>();

    @Bindable
    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
        notifyPropertyChanged(BR.name);
    }

    @Bindable
    public int getNum() {
        return num;
    }

    public void setNum(int num) {
        this.num = num;
        notifyPropertyChanged(BR.num);
    }

    public void onClickUp() {
        this.num = num++;
        observableField.set("Number : " + num);
        notifyPropertyChanged(BR.num);
    }

}

File: MVVMDemo/app/src/main/java/com/example/mvvmdemo/UserViewModel.kt
package com.example.mvvmdemo

import android.util.Log
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import androidx.lifecycle.ViewModel

class UserViewModel : ViewModel() {

    private var _number: MutableLiveData<Int> = MutableLiveData(0)
    val number: LiveData<Int> = _number

    fun setNum(value:Int){
        _number.value = value
    }

    fun onClickUp(){
        _number.value = _number.value?.plus(1)
        Log.e("TAG", "onClickUp: ${_number.value}")
    }
}